tests/infer/test_autoguide.py

```diff
@@ -256,14 +257,28 @@ def model():
     guide()
     if auto_class is AutoLaplaceApproximation:
         guide = guide.laplace_approximation()
+    pyro.set_rng_seed(0)
+    expected = guide.call()
+    names = sorted(guide())
 
     # Ignore tracer warnings
     with warnings.catch_warnings():
         warnings.filterwarnings("ignore", category=torch.jit.TracerWarning)
         # XXX: check_trace=True fails for AutoLaplaceApproximation
         traced_guide = torch.jit.trace_module(guide, {"call": ()}, check_trace=False)
-    torch.jit.save(traced_guide, "/tmp/test_guide_serialization.pt")
-    guide_deser = torch.jit.load("/tmp/test_guide_serialization.pt")
+    f = io.BytesIO()
+    torch.jit.save(traced_guide, f)
+    f.seek(0)
+    guide_deser = torch.jit.load(f)
+
+    # Check .call() result.
+    pyro.set_rng_seed(0)
+    actual = guide_deser.call()
```

Comment:
For a `TorchScript` instance, we will completely bypass the getattr/setattr logic in our `PyroModule` class (it will probably just exercise some python glue in pytorch and call the underlying C++ object). As such, accessing params will not load their values into the param store, and we should be using `pickle` if we would like to reuse serialized guides for training purposes. Is that correct? Just looking to check my understanding.

Reply:
Correct. Actually I wanted to chat with you about that: `pyro.load_param_store()` will no longer work as a method to save `AutoGuide` or `EasyGuide`s. I'm not sure whether we should simply document this or add new options (similar to `pyro.module(..., update_module_params=True)`. Adding as a task to #2120 .